float32 x
float32 y
float32 radius
bool isDetected
bool isStable
